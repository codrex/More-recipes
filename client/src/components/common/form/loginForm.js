import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Form from './form';
import Input from './input';
import { validateLogin } from '../../../validator/validator';

/**
 * User login form
 */
class LoginForm extends React.Component {
  /**
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.login = this.login.bind(this);
  }
   /**
    * @param {object} value (form values)
   * @return {undefined} undefined
  */
  login(value) {
    this.props.login(value);
  }

  /**
   * @returns {object} the form
   */
  render() {
    const { handleSubmit } = this.props;
    return (
      <Form
        submitBtnText="Login"
        onSubmit={handleSubmit(this.props.login)}
        className={this.props.loading ? 'hide' : ''}
        primary
      >
        <Field
          component={Input}
          name="username"
          type="text"
          id="username"
          placeholder="Enter username"
        />
        <Field
          component={Input}
          name="password"
          type="password"
          id="password"
          placeholder="Enter password"
        />
      </Form>
    );
  }
}
LoginForm.propTypes = {
  login: PropTypes.func,
  handleSubmit: PropTypes.func,
  loading: PropTypes.bool,
};

export default reduxForm({
  validate: validateLogin,
  form: 'LoginForm'
})(LoginForm);
