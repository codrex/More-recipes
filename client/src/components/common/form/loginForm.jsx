import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Form from './form';
import Input from './input';
import { username, password } from '../../../utils/validator/validator';

/**
 * User login form
 */
class LoginForm extends React.Component {
  /**
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.login = this.login.bind(this);
  }
   /**
    * @param {object} value (form values)
   * @return {undefined} undefined
  */
  login(value) {
    this.props.login(value);
  }

  /**
   * @returns {object} the form
   */
  render() {
    const { handleSubmit } = this.props;
    return (
      <Form
        submitBtnText="Login"
        onSubmit={handleSubmit(this.props.login)}
        disabled={this.props.loading}
        secondary
      >
        <Field
          component={Input}
          name="username"
          type="text"
          id="username"
          placeholder="Enter username"
          validate={username}
        />
        <Field
          component={Input}
          name="password"
          type="password"
          id="password"
          placeholder="Enter password"
          validate={password}
        />
      </Form>
    );
  }
}
LoginForm.propTypes = {
  login: PropTypes.func,
  handleSubmit: PropTypes.func,
  loading: PropTypes.bool,
};

export default reduxForm({
  form: 'LoginForm'
})(LoginForm);
