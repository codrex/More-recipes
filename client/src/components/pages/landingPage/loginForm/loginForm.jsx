import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Form from '../../../common/form/form';
import Input from '../../../common/form/input';
import { username, password } from '../../../../utils/validator/validator';

/**
 * User login form
 */
export class LoginForm extends React.Component {
  /**
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.login = this.login.bind(this);
  }
   /**
    * @param {object} value (form values)
   * @return {undefined} undefined
  */
  login(value) {
    this.props.login(value);
  }

  /**
   * @returns {object} the form
   */
  render() {
    const { handleSubmit, submitting } = this.props;
    return (
      <Form
        submitBtnText={submitting ? 'loading...' : 'Login'}
        onSubmit={handleSubmit(this.props.login)}
        disabled={submitting}
        secondary
      >
        <Field
          component={Input}
          name="username"
          type="text"
          id="username"
          placeholder="Enter username"
          validate={username}
        />
        <Field
          component={Input}
          name="password"
          type="password"
          id="password"
          placeholder="Enter password"
          validate={password}
        />
      </Form>
    );
  }
}
LoginForm.propTypes = {
  login: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
};

export default reduxForm({
  form: 'LoginForm'
})(LoginForm);
