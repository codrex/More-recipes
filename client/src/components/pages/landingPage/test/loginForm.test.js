import React from 'react';
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../../../../store/configStore';
import { PureLoginForm } from '../loginForm/loginForm';

const props = {
  login: jest.fn(),
  handleSubmit: jest.fn(),
  submitting: false
};

describe('login form component ', () => {
  test('render as expected when submitting props is false', () => {
    const component = shallow(<PureLoginForm {...props} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });
  test('render as expected when submitting props is true', () => {
    const component = (
      <Provider store={configureStore()}>
        <BrowserRouter>
          <PureLoginForm {...{ ...props, submitting: true }} />
        </BrowserRouter>
      </Provider>
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
