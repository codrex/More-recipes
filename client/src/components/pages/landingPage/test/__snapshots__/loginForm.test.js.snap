// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form component  render as expected when submitting props is false 1`] = `
<Form
  action=""
  className=""
  disabled={false}
  lg={true}
  method=""
  onSubmit={[Function]}
  primary={false}
  primaryInverse={false}
  secondary={true}
  secondaryInverse={false}
  submitBtnText="Login"
>
  <Field
    component={[Function]}
    id="username"
    name="username"
    placeholder="Enter username"
    type="text"
    validate={[Function]}
  />
  <Field
    component={[Function]}
    id="password"
    name="password"
    placeholder="Enter password"
    type="password"
    validate={[Function]}
  />
</Form>
`;

exports[`login form component  render as expected when submitting props is true 1`] = `undefined`;
